scio-core/src/main/java/com/spotify/scio/transforms/FutureHandlers.java

```diff
@@ -87,7 +110,7 @@ public void onFailure(Throwable t) {
               }
             }
           },
-          MoreExecutors.directExecutor());
```

Comment:
MoreExecutors.directExecutor() doc says callback may not be executed if the exception terminates the executing thread

> In other cases, no code will catch the exception, and it may terminate whichever thread happens to trigger the execution.

I've changed to use another executor that also completes the future when if it refuses to execute the callback. This is the strategy used by guava when transforming futures.